test/app/core/utils.spec.coffee

```diff
@@ -1,6 +1,158 @@
+        jasmine.clock().mockDate(new Date(2013, 11, 1))
+        expect(utils.yearsSinceMonth('2012-12-01')).toBe(1)
+      it 'Dec 31 2012 to Dec 31 2013 should be 1 year', ->
+        jasmine.clock().mockDate(new Date(2013, 11, 31))
+        expect(utils.yearsSinceMonth('2012-12-31')).toBe(1)
+
+  describe 'ageToBracket', ->
+    describe 'should return open if', ->
+      it 'undefined', ->
+        expect(utils.ageToBracket()).toBe('open')
+      it '>= 19', ->
+        expect(utils.ageToBracket(19)).toBe('open')
+    describe "20-21 school year", ->
+      getAge = (now, birth)->
+        utils.daysBetween(now,new Date(birth))/365.5
+      dates = [
+        '2020-09-02','2020-09-16','2020-09-28',
+        '2020-11-02','2020-11-16','2020-11-28',
+        '2021-01-02','2021-01-16','2021-01-28',
+        '2021-02-02','2021-02-16','2021-02-28',
+        '2021-05-02','2021-05-16','2021-05-28',
+        '2021-06-02','2021-06-16','2021-06-28',
+      ]
+      beforeEach ->
+        jasmine.clock().install()
+      afterEach ->
+        jasmine.clock().uninstall()
+      describe 'older than bracketMax', ->
+        it 'will age in to current bracket AND not aged in to the next one', ->
+          now = new Date('2021-02-25')
+          jasmine.clock().mockDate(now)
+          birthDate = '2009-08-31'
+          age = getAge(now,birthDate)
+          # SEASON 1:
+          # age: 4196, seasonLength: 119, bracketMax: 4140, daysOlderThanBracketMax: 56, daysElapsedFromSeason: 55
+          expect(utils.ageToBracket(age)).toBe('0-11')
+        it 'will age in to current bracket BUT already aged in to the next one', ->
+          now = new Date('2021-02-25')
+          jasmine.clock().mockDate(now)
+          birthDate = '2009-09-01'
+          age = getAge(now,birthDate)
+          # SEASON 1:
+          # age: 4195, seasonLength: 119, bracketMax: 4140, daysOlderThanBracketMax: 55, daysElapsedFromSeason: 55}
+          # SEASON 2:
+          # age: 4195, seasonLength: 119, bracketMax: 5236, daysOlderThanBracketMax: -1041, daysElapsedFromSeason: 55}
+          expect(utils.ageToBracket(age)).toBe('11-14')
+      describe 'should be in 0-11 bracket', ->
+        it 'if born at 9/1/2014', ->
+          birthDate = '2014-09-01'
+          for date in dates
+            now = new Date(date)
+            jasmine.clock().mockDate(now)
+            expect(utils.ageToBracket(getAge(now,birthDate))).toBe('0-11')
+        it 'if born at 8/31/2013', ->
+          birthDate = '2013-08-31'
+          for date in dates
+            now = new Date(date)
+            jasmine.clock().mockDate(now)
+            expect(utils.ageToBracket(getAge(now,birthDate))).toBe('0-11')
+        it 'if born at 9/1/2010', ->
+          birthDate = '2010-09-01'
+          for date in dates
+            now = new Date(date)
+            jasmine.clock().mockDate(now)
+            expect(utils.ageToBracket(getAge(now,birthDate))).toBe('0-11')
+        it 'if born at 8/31/2009', ->
```

Comment:
a child born at 8/31 should be older than the child born at 9/01 then we cannot say the older child is 0-11 and the younger is 11-14